### DISCUSSION TITLE
Can it run KOTOR?

### DISCUSSION THREAD

**Discussion by TerrorizedGakon** [pal1000/mesa-dist-win#190] [Q&A]
Straightforward question. Can this be used to run an OpenGL game like KOTOR from 2003? On a modern AMD GPU.

**Comment by pal1000** [comment]
It's worth trying.

**Reply by TerrorizedGakon** [reply]
I tried and the game is crashing instantly.

How can I troubleshoot?

I tried MESA_EXTENSION_MAX_YEAR=2003, MESA_GL_VERSION_OVERRIDE=3.3COMPAT, didn't make a difference.

Do you think your build could help with graphical glitches this game has on Windows? On Linux the game works fine using Steam proton.

**Reply by pal1000** [reply]
Since you are on modern AMD GPU you can try all Mesa3D gallium drivers. Try setting these environment variables 1 by 1:
````
GALLIUM_DRIVER   d3d12
GALLIUM_DRIVER   zink
GALLIUM_DRIVER   llvmpippe
````
d3d12 is default if GALLIUM_DRIVER is unset.
This should narrow which driver(s) is/are broken.

**Reply by TerrorizedGakon** [reply]
Here is a batch file I have tried

set MESA_LOG_FILE=D:\SteamLibrary\steamapps\common\swkotor\swkotor.txt
set MESA_DEBUG=1
set GALLIUM_DRIVER=llvmpipe
set MESA_GL_VERSION_OVERRIDE=3.2
set MESA_GLSL_VERSION_OVERRIDE=150
set MESA_EXTENSION_MAX_YEAR=2003
swkotor.exe

I have tried all gallium drivers, d3d12, zink, llvmpipe. The game starts up and then after a few seconds closes down with no message whatsoever, no matter which is used.

I'm using the MSVC package and per-app deployment

**Reply by pal1000** [reply]
Take out MESA_GL_VERSION_OVERRIDE and MESA_GLSL_VERSION_OVERRIDE, kotor is definitely too old to need them.
Also please [read this](https://steamcommunity.com/app/32370/discussions/0/1620599015890067721/).
